libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -593,6 +593,65 @@ std::string YulUtilFunctions::resizeDynamicArrayFunction(ArrayType const& _type)
 	});
 }
 
+string YulUtilFunctions::storageArrayPopFunction(ArrayType const& _type)
+{
+	solAssert(_type.location() == DataLocation::Storage, "");
+	solAssert(_type.isDynamicallySized(), "");
+	solUnimplementedAssert(!_type.isByteArray(), "Byte Arrays not yet implemented!");
+	solUnimplementedAssert(_type.baseType()->storageBytes() <= 32, "...");
+	solUnimplementedAssert(_type.baseType()->storageSize() == 1, "");
+
+	string functionName = "array_pop_" + _type.identifier();
+	return m_functionCollector->createFunction(functionName, [&]() {
+		return Whiskers(R"(
+			function <functionName>(array) {
+				let oldLen := <fetchLength>(array)
+				if iszero(oldLen) { invalid() }
+				let newLen := sub(oldLen, 1)
+
+				let slot, offset := <indexAccess>(array, newLen)
+				<setToZero>(slot, offset)
+
+				sstore(array, newLen)
+			})")
+			("functionName", functionName)
+			("fetchLength", arrayLengthFunction(_type))
+			("indexAccess", storageArrayIndexAccessFunction(_type))
+			("setToZero", storageSetToZeroFunction(*_type.baseType()))
+			("maxArrayLength", (u256(1) << 64).str())
+			.render();
+	});
+}
+
+string YulUtilFunctions::storageArrayPushFunction(ArrayType const& _type)
+{
+	solAssert(_type.location() == DataLocation::Storage, "");
+	solAssert(_type.isDynamicallySized(), "");
+	solUnimplementedAssert(!_type.isByteArray(), "Byte Arrays not yet implemented!");
+	solUnimplementedAssert(_type.baseType()->storageBytes() <= 32, "...");
```

Comment:
```suggestion
	solUnimplementedAssert(_type.baseType()->storageBytes() <= 32, "");
```

Reply:
I've also cleaned up this and the line below.